cockatrice/src/window_main.cpp

```diff
@@ -288,6 +293,87 @@ void MainWindow::actAbout()
     ));
 }
 
+void MainWindow::actUpdate()
+{
+    //Request the newest builds
+    QUrl versionUrl(LATEST_FILES_URL);
+    response = netMan.get(QNetworkRequest(versionUrl));
+    connect(response, SIGNAL(finished()),
+            this, SLOT(fileListFinished()));
+}
+
+void MainWindow::fileListFinished() {
+    //Parse the version results into a JSON array
+    QString strReply = (QString) response->readAll();
+    QJsonDocument jsonResponse = QJsonDocument::fromJson(strReply.toUtf8());
+    QJsonArray json_array = jsonResponse.array();
+
+    //From this array we intend to find the correct downloadUrl, if needed
+    QString downloadUrl;
+    bool correctVersion = false, needToUpdate = false;
+    QString dateStr;
+    int downloadSize;
+
+    //Iterate over the builds
+    foreach (const QJsonValue &value, json_array)
+    {
+        //Find the build's name, date, and URL
+        QJsonObject json_obj = value.toObject();
+        QString name = json_obj["name"].toString();
+        dateStr = json_obj["created"].toString();
+        downloadSize = json_obj["size"].toInt();
+
+        //Parse the date
+        QString formatString = "yyyy-MM-dd";
+        QDate date = QDate::fromString(dateStr.remove(formatString.length(), 50), formatString);
+
+        //If we ever find a build for our OS, set correctVersion to true
+#if defined(Q_OS_OSX)
+        if(name.contains("osx"))
+            correctVersion = true;
+#elif defined(Q_OS_WIN)
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+        if(name.contains("qt5.exe"))
+            correctVersion = true;
+#else
+        if(name.contains("qt4.exe"))
+            correctVersion = true;
+#endif
+#endif
+            //If there is ever a file that's newer than ours, we need to update even if we can't
+            if(date > lastCommit)
+                needToUpdate = true;
+
+            //If we've found the correct version, get the URL and stop reading versions
+            if (correctVersion)
+            {
+                downloadUrl = DOWNLOAD_URL + json_obj["path"].toString();
+                break;
+            }
+        }
+
+    //If there's something to download, prompt the user
+    if (needToUpdate)
+    {
+        if (correctVersion) {
+            QMessageBox::StandardButton reply;
+            reply = QMessageBox::question(this, "Update Cockatrice",
```

Comment:
Can the git commit of the new version as well as the commit+date of the current version be included in this message please?